### PR TITLE
Tweak: Make the section element extendable

### PR DESCRIPTION
**Pull request by Oxibug**
## PR Checklist

- [ x ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ x ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary
Closed PR #13449 Before forking

Give the developers the ability to customize the main section so they can use the main section background widely, like adding Canvas backgrounds instead of Videos or classic BG and it will be great to use the section background in anyway they wish.

## Description
Extend Element_Section class using action [elementor/elements/elements_registered] to add a new HTML output in the section's background such as using Canvas instead of Videos background,

## Test instructions
This PR can be tested by following these steps:

* In function [init_elements] located in Elements_Manager class, I need you to add **'$this'** word to action 'elementor/elements/elements_registered' So I'll be able to use the current class's function [register_element_type] .. like that:
**do_action( 'elementor/elements/elements_registered', $this );**

* Modify functions [**get_html_tag**] and [**print_shape_divider**] located in **Element_Section** class in [section.php] to public instead of private.

* Developer can add new controls using action '**elementor/element/section/section_advanced/after_section_end**' mentioned in the opened issue [3334](https://github.com/elementor/elementor/issues/3334#issuecomment-361611013)

* Now the developer can Extends the **Element_Section** class and override the functions [**content_template**], [**before_render**], [**after_render**]

## Quality assurance

- [ x ] I have tested this code to the best of my abilities
- [ x ] I have added unittests to verify the code works as intended
- [ x ] Docs have been added / updated (for bug fixes / features)

Fixes #

* Now, The developer can copy the three function [**content_template**], [**before_render**], [**after_render**] from **\elementor\includes\elements\section.php**
* Use the action [**elementor/elements/elements_registered**] after adding **$this** to point to the parent class and extend his new section **with the same implementation of the Elementor Plugin's Section** but now he can add his own HTML output as he wish

```
class ModifyElementorSection {

    public function __construct() {
        add_action( 'elementor/elements/elements_registered',         [ $this, 'modify_section' ] );
    }

    public function modify_section( $el_manager ) {        

        $el_manager->register_element_type( new Customizable_Section() );

    }

}

class Customizable_Section extends \Elementor\Element_Section {

    public function content_template() {
        /* Copy the Elementor's section form the Plugin Then append your new HTML output */
    }

    public function before_render() {
        /* Copy the Elementor's section form the Plugin Then append your new HTML output */
    }

    public function after_render() {
        /* Copy the Elementor's section form the Plugin Then append your new HTML output */
    }

}
```

### CODE REVIEW

**Comment by Oxibug** [comment]
@shilo-ey
Hi There, It seems like you forget the request, waiting for more than a month :) 
Is that feature possible? I made my own Elementor for just one word

**Comment by Oxibug** [comment]
@shilo-ey 
Hi Pal, Can you look into this PR because I'm waiting for 6 months and after merging the new version I saw this msg [Block non-permitted User] 

I think I need to be approved manually from an authorized author, and the others [Waiting for status to be reported] msgs I don't know what should I do, I jus added onw word in the entire project.

I made a search and make an empty commit but it getting worst instead of 2 warnings I have 7 now :)

File: includes/elements/section.php [hunk 1]
```diff
@@ -1560,12 +1560,14 @@ protected function _get_default_child_type( array $element_data ) {
 	 *
 	 * Retrieve the section element HTML tag.
 	 *
+	 * Modifed by Oxibug - NULL Commit
```
Comment (shilo-ey) [inline includes/elements/section.php:1563]: Please remove this line

File: includes/elements/section.php [hunk 2]
```diff
@@ -1580,12 +1582,14 @@ private function get_html_tag() {
 	 *
 	 * Used to generate the shape dividers HTML.
 	 *
+	 * Modifed by Oxibug - NULL Commit
```
Comment (shilo-ey) [inline includes/elements/section.php:1585]: Please remove this line

File: includes/managers/elements.php [hunk 3]
```diff
@@ -252,10 +252,12 @@ private function init_elements() {
 		 * After elements registered.
 		 *
 		 * Fires after Elementor elements are registered.
+		 * 
+		 * Modified by Oxibug - NULL Commit
```
Comment (shilo-ey) [inline includes/managers/elements.php:256]: Please remove this line

**Comment by shilo-ey** [comment]
Hey @Oxibug 
Thank you for your contribution

Please remove the credit lines from the code itself. Elementor is an open source project and credit will be given in the Changelog.

Thanks!

**Review by shilo-ey — CHANGES_REQUESTED** [review]

**Review by matipojo — APPROVED** [review]

File: includes/elements/section.php [hunk 4]
```diff
@@ -1565,7 +1565,7 @@ protected function _get_default_child_type( array $element_data ) {
 	 *
 	 * @return string Section HTML tag.
 	 */
-	private function get_html_tag() {
+	public function get_html_tag() {
```
Comment (matipojo) [inline includes/elements/section.php:1568]:
```suggestion
	protected function get_html_tag() {
```

File: includes/elements/section.php [hunk 5]
```diff
@@ -1585,7 +1585,7 @@ private function get_html_tag() {
 	 *
 	 * @param string $side Shape divider side, used to set the shape key.
 	 */
-	private function print_shape_divider( $side ) {
+	public function print_shape_divider( $side ) {
```
Comment (matipojo) [inline includes/elements/section.php:1588]:
```suggestion
	protected function print_shape_divider( $side ) {
```

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by matipojo — APPROVED** [review]

**Comment by danielezerosi** [comment]
@shilo-ey
@kobizz

Please approve the changes as this customization is very important to many developers

**Review by shilo-ey — APPROVED** [review]

**Review by kobizz — APPROVED** [review]

**Pull request merged**